rust/template/distributed_datalog/src/tcp_channel/receiver.rs

```diff
@@ -222,22 +315,13 @@ where
         &mut self,
         observer: ObserverBox<T, String>,
     ) -> Result<Self::Subscription, ObserverBox<T, String>> {
-        let mut guard = self.observer.lock().unwrap();
-        if guard.is_some() {
-            Err(observer)
-        } else {
-            guard.replace(observer);
-            Ok(())
-        }
+        self.txnmux.lock().unwrap().subscribe(observer)
```

Comment:
Should this be unsubscribe?

Reply:
Not following. Why would it be `unsubscribe`? This is the implementation of the `subscribe` method, after all, and it just forward the request to the `txnmux`.

Reply:
Nevermind.  I got confused by how the diff was displayed. My bad.